ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committing/ProcessTransactionSubmission.scala

```diff
@@ -400,7 +414,12 @@ private[kvutils] class ProcessTransactionSubmission(
           key ->
             DamlStateValue.newBuilder
               .setContractKeyState(
-                DamlContractKeyState.newBuilder.setContractId(contractKeyState.fold("")(_.coid))
+                contractKeyState
```

Comment:
To improve readability, please move this into a helper function like `updateContractKeyWithContractKeyState(...)`.

Reply:
Fixed in 14a595f.